### PR TITLE
feat: add support for RoleARN and Filters in EC2SDConfigs

### PR DESCRIPTION
**Pull request by slashpai**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Related-to https://github.com/prometheus-operator/prometheus-operator/issues/5818

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
feat: add support for RoleARN and Filters in EC2SDConfigs
```

### CODE REVIEW

File: pkg/prometheus/testdata/ScrapeConfigSpecConfig_EC2SDConfigValidAPIKeys.golden
Comment (ArthurSens) [inline pkg/prometheus/testdata/ScrapeConfigSpecConfig_EC2SDConfigValidAPIKeys.golden:1]: I understand you're just renaming the file here but I can't help but notice that the file is completely empty. Should it be?

**Review by ArthurSens — APPROVED** [review]
Code changes LGTM, but I've noticed something strange regarding an already existing test case

File: pkg/prometheus/testdata/ScrapeConfigSpecConfig_EC2SDConfigValidAPIKeys.golden
Comment (slashpai) [reply pkg/prometheus/testdata/ScrapeConfigSpecConfig_EC2SDConfigValidAPIKeys.golden:1]: I am not sure why GitHub doesn't show file correctly but this is the file from branch https://github.com/slashpai/prometheus-operator/blob/ec2_sd/pkg/prometheus/testdata/ScrapeConfigSpecConfig_EC2SDConfigValidAPIKeys.golden

**Review by xiu — APPROVED** [review]
LGTM

**Pull request merged**